Microsoft.Toolkit.Uwp.UI.Controls.Media/ImageCropper/ImageCropper.Logic.cs

```diff
@@ -23,6 +23,12 @@ public partial class ImageCropper
         /// <param name="animate">Whether animation is enabled.</param>
         private void InitImageLayout(bool animate = false)
         {
+            if (!IsValidRect(CanvasRect))
+            {
+                _lazyInitImageLayoutAction = () => InitImageLayout(animate);
```

Comment:
Something about this feels off...

@HHChaos were you able to pinpoint exactly what caused the original issue? Could we get some additional insight into the cause and the fix?

Reply:
I just noticed we don't do any gate on the control to be loaded. Maybe that'd be the better fix is waiting for layout to complete?

Reply:
@Arlodotexe The source of this bug is when ImageCropper is in the layout arranging phase, its layout may take more time if the available sizes are infinite, and because CanvasRect is not a valid size before the layout is completed, this bug may occur if the InitImageLayout method is called at this time.

@michael-hawker I have tested it, when loaded occurs, CanvasRect may still be an invalid value.

Reply:
This fix feels like it's patching over a slightly deeper issue.

Looking at the code, it appears that
- We're relying on the size of the `PART_ImageCanvas` that contains the image, rather than the image itself.
- We're calling `InitImageLayout` when the source has changed, but not when the image is loaded.

If the above ends up being the cause of other issues down the road, we don't want to be band-aid patching them. When underlying issues get fixed that have been partially patched over, we need to clean up all those patches, and there are no notes or comments to tell us what code should be included in the cleanup.

To save time and headache later, I think we should look at fixing this by fixing the underlying problem.

Reply:
Looks like the source of the problem is that calling `await imgCropper.LoadImageFromFile(file);` doesn't wait for the image to load before completing.

Adding a `Task.Delay()` with any value (minimum of 1 works) fixes the issue.

Notably, this issue also exists when do you
```csharp
imgCropper.Source = writableBitmap;
imgCropper.AspectRatio = 1;
```

Reply:
Additional information
- When setting the Source property, CanvasRect has a width but no height, causing it to be treated as invalid.
  - Presumably, this is because the image is not yet loaded and the containing Grid row uses `Auto` for a height.
  - Assigning a `Height` to the ImageCropper is a workaround (not a fix)

Reply:
PR #4720 opened with a much simpler fix